src/parser.rs

```diff
@@ -3652,7 +3652,10 @@ impl<'a> Parser<'a> {
         &mut self,
         optional: IsOptional,
     ) -> Result<Vec<Ident>, ParserError> {
-        if self.consume_token(&Token::LParen) {
+        if self.peek_token() == Token::LParen
```

Comment:
@mobuchowski could you please:
- Add an example of which dialect supports that syntax
- Make sure this doesn't break another syntax. Is using `SELECT` the best approach here? It seems too manual and prone to problems, as nested queries could have `WITH` clauses. Am I missing something?

Reply:
Interestingly, it seems like postgres expects a `column_name` https://www.postgresql.org/docs/current/sql-insert.html

So I echo @AugustoFKL 's question about what dialect supports this construction? It is probably good to mention this in comments

In terms of not breaking other dialects, the coverage seems reasonable 

https://github.com/sqlparser-rs/sqlparser-rs/pull/667/files#diff-ae2dab2c924f4018c1daf0743b8fd912c44c5b63ae7144dd8b951bb3bd254349R64